scala/scala.bzl

```diff
@@ -376,32 +377,58 @@ def collect_srcjars(targets):
             srcjars += [target.srcjars.srcjar]
     return srcjars
 
-def update_jars_to_labels(jars2labels, target, all_jars):
+def add_labels_of_jars_to(jars2labels, dependency, all_jars):
   for jar in all_jars:
-    if jar.path not in jars2labels:
-      if hasattr(target, "jars_to_labels") and jar.path in target.jars_to_labels:
-        jars2labels[jar.path] = target.jars_to_labels[jar.path]
-      else:
-        jars2labels[jar.path] = target.label
+   add_label_of_jar_to(jars2labels, dependency, jar)
 
-def _collect_jars(targets):
+
+def add_label_of_jar_to(jars2labels, dependency, jar):
+ if label_already_exists(jars2labels, jar):
+   return
+
+ # skylark exposes only labels of direct dependencies.
+ # to get labels of indirect dependencies we collect them from the providers transitively
+ if provider_of_dependency_contains_label_of(dependency, jar):
+   jars2labels[jar.path] = dependency.jars_to_labels[jar.path]
+ else:
+   jars2labels[jar.path] = dependency.label
+
+
+def label_already_exists(jars2labels, jar):
+  return jar.path in jars2labels
+
+def provider_of_dependency_contains_label_of(dependency, jar):
+ return hasattr(dependency, "jars_to_labels") and jar.path in dependency.jars_to_labels
+
+def dep_target_contains_ijar(dep_target):
+  return hasattr(dep_target, 'scala') and hasattr(dep_target.scala, 'outputs') and hasattr(dep_target.scala.outputs, 'ijar')
+
+def _collect_jars(dep_targets):
     """Compute the runtime and compile-time dependencies from the given targets"""  # noqa
     compile_jars = depset()
     runtime_jars = depset()
+    transitive_cjars = depset()
     jars2labels = {}
-    for target in targets:
-        if java_common.provider in target:
-            java_provider = target[java_common.provider]
+    for dep_target in dep_targets:
+        if dep_target_contains_ijar(dep_target):
+          transitive_cjars += [dep_target.scala.outputs.ijar]
+        if hasattr(dep_target, 'transitive_cjars'):
+          transitive_cjars += dep_target.transitive_cjars
+        if java_common.provider in dep_target:
+            java_provider = dep_target[java_common.provider]
             compile_jars += java_provider.compile_jars
+            transitive_cjars += java_provider.compile_jars
```

Comment:
[Java provider](https://github.com/bazelbuild/bazel/blob/77baa4c372ae014db8e17f6d3c60bd88597853d9/src/main/java/com/google/devtools/build/lib/rules/java/JavaProvider.java#L177) only exposes `compile_jars` and `transitive_runtime_jars`.
I've opened an [issue](https://github.com/bazelbuild/bazel/issues/3367) for transitive_compile_time_jars to be exposed